### ISSUE TITLE
Make project_role an ontology field

### ISSUE THREAD

**Issue by malloryfreeberg** [HumanCellAtlas/metadata-schema#894]
**For which schema is a change/update being suggested?**

I would like to request an update to the `contact.json` schema.

**What should the change/update be?**

I would like to change this field from an enum to an official ontology. This ticket is currently blocked by adding the new `project_role_ontology.json` schema (issue #893).

**What new field(s) need to be changed/added?**

No new fields.

**Why is the change requested?**

Governing the `project_role` field by an ontology instead of an enum is more robust to future updates. Linked to issue #855 which states (from Dani) "All terms apart from data wrangler added/updated in EFO. The next EFO release will be on Monday [18/03/19] and I will do an HCA-internal release in the next couple of weeks, subject to resolving some unrelated blocking ontology issues."

**Comment by malloryfreeberg** [comment]
@ESapenaVentura if you want to try a more complex schema update, feel free to take this one :)

**Comment by ESapenaVentura** [comment]
@malloryfreeberg will do! I'm actually really curious about this task, so I'm looking forward to do it!

**Comment by ESapenaVentura** [comment]
@malloryfreeberg I know that this is a more robust way of writing this field for future updates, but shouldn't the ontology support the term "other"? I'm not quite sure if this topic should be here, but I think that updating it without giving the opportunity for data contributors to put an EFO inside organization role for "others" (e.g., they don't feel the current terms fit the description of their job) could limit the process of getting new data. But I'm quite new, so I would be really glad to have your feedback in here. Thanks!

**Comment by daniwelter** [comment]
Ahhh @ESapenaVentura, looks like someone needs their Ontologies 101 session soon.. ;-)

**Comment by ESapenaVentura** [comment]
@daniwelter Yes, please! I know pretty much only basic things relating the ontologies field, and by the look of your message, I have made a horrible mistake in my previous message!

**Comment by malloryfreeberg** [comment]
@ESapenaVentura not a horrible mistake :) That's a perfectly reasonable question about having the option for "other". But definitely have a discussion with @daniwelter about this (and other) ontology topics. The tl;dr is that we are able to extend the ontology if someone feels the value they want to add isn't reflected in the current ontology. There are many reasons to want to avoid a catch-all "other" category, e.g. it gives people an easy way to enter a value without having to think about whether the value they want is in the ontology :)

**Comment by ESapenaVentura** [comment]
@malloryfreeberg Oh, ok, now I understand! Yes, I have quick talk on the onboard meetings with @daniwelter , so I hope to get into further details soon!

**Comment by ESapenaVentura** [comment]
I pushed what I think are all the necessary changes. However, I need to fix my 'jsonschema' module on python because it is not able to import 'Draft7Validator' and so, I can't run the `schemas_are_valid_json.py` to verify the updates in the metadata schema. I'll try to fix it this weekend!

**Comment by malloryfreeberg** [comment]
I think we can actually deprecate back to the Draft4Validator. The Draft7Validator isn't released yet, and the Draft4Validator will actually work fine.

Actually what we need to do is implement the javascript json validator, as that is the one that ingest actually uses. Maybe @ESapenaVentura you can work with @simonjupp or the ingest developers to get this working (not a priority right now, though).

In the meantime, you can change Draft7Validator to Draft4Validator in the python script locally and run it.
